flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskTest.java

```diff
@@ -871,6 +901,9 @@ public void testTerminationFutureCompletesOnNormalExecution() throws Exception {
 
 		assertFalse(task.getTerminationFuture().isDone());
 
+		task.getExecutingThread().join(500); // allow some arbitrary delay for potential termination and re-check
+		assertFalse(task.getTerminationFuture().isDone());
```

Comment:
I'm not sure if I understand this part. Was this 500ms delay inducing a failure in this test in the previous version if this PR? If not, I don't think adding such sleep would be justified.

Reply:
Removed.

Reply:
Removed.